tofhir-engine/src/main/scala/io/tofhir/engine/model/DataProcessingSettings.scala

```diff
@@ -0,0 +1,21 @@
+package io.tofhir.engine.model
+
+import io.tofhir.engine.config.ErrorHandlingType.ErrorHandlingType
+import io.tofhir.engine.model.ArchiveModes.ArchiveModes
+
+object ArchiveModes extends Enumeration {
+  type ArchiveModes = String
+  final val OFF = "off"
+  final val DELETE = "delete"
+  final val ARCHIVE = "archive"
+}
+
+/**
+ * Data source processing settings
+ * @param mappingErrorHandling Error handling type HALT or CONTINUE
+ * @param saveErroneousRecords If true, erroneous records will be saved to archive folder with the same path as the input file
```

Comment:
Do we have an archive folder which keeps the archived source files as well as the erroneous records? How do we differentiate them, by subfolders?

I see an error-folder-path configuration, is that under archive folder path config?

Reply:
We have two seperate configs: one for erroneous records and one for archiving. These configs specify the folder names for each and they work independently. Let's say these configs are set as following:
erroneousRecordsFolder: erroneous-records-folder
archiveFolder: archive-folder

The path structure of erroneous records:
`erroneous-records-folder/<error-type>/job-<jobId>/execution-<executionId>/<mappingUrl>/<random-generated-name-by-spark>.csv`

In the archiving mechanism, we use the same path of the file source relative the to repository (context) and put it in the configured archiving folder without changing the path. 
If path of csv file is this: `C:/dev/be/data-integration-suite/test-data/streaming-folder/patients/patients.csv` and context path is this: `C:/dev/be/data-integration-suite` 
We get the relative path of data source to the context path and process archiving according to it. 
Archiving path: `archive-folder/test-data/streaming-folder/patients/patients.csv`
That way, we can understand from which location the data source is archived.